api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java

```diff
@@ -387,18 +406,181 @@ public boolean supportsPropertyName(String gpName) {
+	public Patient importMpiPatient(String patientIdentifier, String patientIdentifierTypeUuid) {
+		MpiPatient foundPatient = findMpiPatient(patientIdentifier, patientIdentifierTypeUuid);
+
+		Patient savedPatient = null;
+		try {
+			if (foundPatient == null) {
+				throw new MpiException(String.format(
+						"Error during importing Patient from MPI. "
+								+ "Patient ID: %s of identifier type: %s has not been found in MPI",
+						patientIdentifier, patientIdentifierTypeUuid));
+			}
+			savedPatient = persistImportedMpiPatient(foundPatient);
+		} catch (RuntimeException e) {
+			servePdqExceptionAndThrowAgain(e,
+					prepareParameters(patientIdentifier, patientIdentifierTypeUuid),
+					PdqErrorHandlingService.PERSIST_MPI_PATIENT_DESTINATION);
+		}
+		return savedPatient;
+	}
+
+	private void servePdqExceptionAndThrowAgain(RuntimeException e, String message,
+			String destination) {
+		ErrorHandlingService errorHandler = registrationCoreProperties.getPdqErrorHandlingService();
+		if (errorHandler == null) {
+			throw new MpiException(message + " with not configured PDQ error handler", e);
 		} else {
-			//should not pass here since "importPatient" performs only when MpiProvider is not null
-			throw new APIException("Should not perform 'fetchMpiPatient' when MPI is disabled");
+			errorHandler.handle(message, destination,true,
+					ExceptionUtils.getFullStackTrace(e));
+			throw new MpiException(message, e);
+		}
+	}
+
+	private String prepareParameters(String identifier, String identifierTypeUuid) {
+		FetchingMpiPatientParameters parameters = new FetchingMpiPatientParameters(identifier,
+				identifierTypeUuid);
+		try {
+			return new ObjectMapper().writeValueAsString(parameters);
+		} catch (IOException e) {
+			throw new RuntimeException("Cannot prepare parameters for OutgoingMessageException", e);
+		}
+	}
+
+	@Override
+	public Patient findByPatientIdentifier(String patientIdentifier, String patientIdentifierTypeUuid) {
+		PatientIdentifierType idType = patientService.getPatientIdentifierTypeByUuid(patientIdentifierTypeUuid);
+		if (idType == null) {
+			throw new APIException(String.format("PatientIdentifierType is missing: UUID %s",
+					patientIdentifierTypeUuid));
+		} else {
+			//Currently method getPatients() does not take into consideration the identifierTypes,
+			//so it needs to be filtered anyway
+			List<Patient> patients = new ArrayList<Patient>();
+			for (Patient p : patientService.getPatients(null, patientIdentifier,
+					Collections.singletonList(idType), true)) {
+				for (PatientIdentifier pi : p.getIdentifiers()) {
+					if (pi.getIdentifierType().equals(idType)) {
+						patients.add(p);
+						break;
+					}
+				}
+			}
+
+			if (CollectionUtils.isEmpty(patients)) {
+				return null;
+			} else {
+				Patient patient = patients.get(0);
+				dao.getSessionFactory().getCurrentSession().refresh(patient);
+				return patient;
+			}
+			//TODO Add error if more than one patient is has that identifier!
```

Comment:
Just replace this by throwing an exception

Reply:
Deleted method, see above